### PR TITLE
test: Remove special case for host identity when remote-node identity is disabled 

### LINKED ISSUE #15491: test: Reconsider special case for host identity when remote-node identity is disabled
We should check if this special case is still needed with latest Cilium:
https://github.com/cilium/cilium/blob/fce7fe070c14d09adf79395106b056111ad86caa/test/k8sT/Policies.go#L1632-L1640

### PR DESCRIPTION
**Pull request by romanspb80**
The test-case "Installing fromEntities host and world policy" with disabled remote-node identity is no longer needed due to lack of any network endpoints outside of the cluster.

Fixes: #15491

Signed-off-by: Roman Ptitcyn romanspb@yahoo.com

### CODE REVIEW

**Review by pchaigno — APPROVED** [review]

**Comment by pchaigno** [comment]
test-me-please

**Comment by romanspb80** [comment]
I think the tests are stuck

**Comment by christarazi** [comment]
test-1.20-4.19

**Comment by christarazi** [comment]
test-1.21-4.9

**Review by jrajahalme — APPROVED** [review]

**Comment by joestringer** [comment]
Was there a discussion about this anywhere? I don't see a detailed analysis either in the issue #15491 or here in this PR.

> The test-case "Installing fromEntities host and world policy" with disabled remote-node identity is no longer needed due to lack of any network endpoints outside of the cluster.

From what I can tell, this test was about policy handling for remote nodes _in_ the cluster, where the "world" entities policy is expected to allow traffic directly from remote nodes in native mode + remote-node-identity=disabled.

Can you explain further what you mean by it is no longer needed? Do we no longer want to make such a guarantee? Or you think that this behaviour is already covered by another test in the testsuite?

**Review by joestringer — CHANGES_REQUESTED** [review]
Please provide a more concrete rationale before we merge this (see comment above).

**Comment by joestringer** [comment]
From out-of-band discussion with @pchaigno it sounds like basically this test code being removed is either (1) not run in regular test environments or (2) not effective in evaluating whether the remote-node identity option is performing correctly, so the extra test code is unnecessary.

@romanspb80 would you mind updating the commit message to include the above detail as well? If for whatever reason we find out in future that this test code was actually useful or necessary, it could be helpful to be able to read back through the git commit logs to find these details out.

With that I think this should be OK to merge.

**Comment by romanspb80** [comment]
> @romanspb80 would you mind updating the commit message to include the above detail as well? If for whatever reason we find out in future that this test code was actually useful or necessary, it could be helpful to be able to read back through the git commit logs to find these details out.
> 
> With that I think this should be OK to merge.

OK. Done.
Thanks

**Comment by joestringer** [comment]
:+1: Only change was to update the commit message, good to merge.

@romanspb80 just to let you know in future, if you configure your email / name in git using the following commands, then you can use `git commit -s` to sign off your commits, so then you can get the standard formatting of the signoff line without needing to manually type out your name / email each time:

```
git config --global user.name "FIRST_NAME LAST_NAME"
git config --global user.email "MY_NAME@example.com"
```

**Review by joestringer — APPROVED** [review]

**Pull request merged**
